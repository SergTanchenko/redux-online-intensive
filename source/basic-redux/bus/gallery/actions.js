// Types
import { SHOW_NEXT_PHOTO, SHOW_SELECTED_PHOTO, SHOW_PREV_PHOTO } from "./types";

export const showPrevPhoto = {
    type: SHOW_PREV_PHOTO,
};

export const showNextPhoto = {
    type: SHOW_NEXT_PHOTO,
};

export const showSelectedPhoto = (selectedPhotoIndex) => {
    return {
        type:    SHOW_SELECTED_PHOTO,
        payload: selectedPhotoIndex,
    };
};
