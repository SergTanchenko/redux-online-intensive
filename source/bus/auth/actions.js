import { types } from "./types";

export const authActions = {
    signupAsync: (userData) => {
        return {
            type:    types.SIGNUP_ASYNC,
            payload: userData,
        };
    },
    authenticate: () => {
        return {
            type: types.AUTHENTICATE,
        };
    },
};
