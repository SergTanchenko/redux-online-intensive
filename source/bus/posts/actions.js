import {
    FILL_POSTS,
    FETCH_POSTS_ASYNC,
    CREATE_POST,
    CREATE_POST_ASYNC
} from "./types";
import { api } from "../../REST/api";

export const fillPosts = (posts) => {
    return {
        type:    FILL_POSTS,
        payload: posts,
    };
};

export const fetchPostsAsync = () => async (dispatch) => {
    dispatch({
        type: FETCH_POSTS_ASYNC,
    });

    const response = await api.posts.fetch();
    const result = await response.json();

    dispatch(fillPosts(result.data));
};

export const createPost = (post) => {
    return {
        type:    CREATE_POST,
        payload: post,
    };
};

export const createPostAsync = (comment) => {
    return {
        type:    CREATE_POST_ASYNC,
        payload: comment,
    };
};
