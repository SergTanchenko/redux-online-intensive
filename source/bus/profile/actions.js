import { types } from "./types";

export const profileActions = {
    // Sync
    fillProfile: (profile) => {
        return {
            type:    types.FILL_PROFILE,
            payload: profile,
        };
    },
    clearProfile: () => {
        return {
            type: types.CLEAR_PROFILE,
        };
    },
    updateAvatar: (avatarUrl) => {
        return {
            type:    types.UPDATE_AVATAR,
            payload: avatarUrl,
        };
    },

    // Async
    updateNameAsync: (newName) => {
        return {
            type:    types.UPDATE_NAME_ASYNC,
            payload: newName,
        };
    },
    updateAvatarAsync: (newAvatar) => {
        return {
            type:    types.UPDATE_AVATAR_ASYNC,
            payload: newAvatar,
        };
    },
    updatePasswordAsync: (passwordInfo) => {
        return {
            type:    types.UPDATE_PASSWORD_ASYNC,
            payload: passwordInfo,
        };
    },
};
