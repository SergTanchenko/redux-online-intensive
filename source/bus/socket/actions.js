import { socket } from "../../init/socket";
import { uiActions } from "../ui/actions";
import { postActions } from "../posts/actions";

export const socketActions = {
    listenConnection: () => (dispatch) => {
        socket.on("connect", () => {
            dispatch(uiActions.setOnlineState());
        });
        socket.on("disconnect", () => {
            dispatch(uiActions.setOfflineState());
        });
    },
    listenPosts: () => (dispatch) => {
        socket.on("create", (event) => {
            const { data: post } = JSON.parse(event);

            dispatch(postActions.createPost(post));
        });
    },
};
