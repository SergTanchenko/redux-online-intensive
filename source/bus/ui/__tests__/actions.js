import { uiActions } from "./../actions";

describe("ui actions: ", () => {
    test("startFetching", () => {
        expect(uiActions.startFetching()).toMatchSnapshot();
    });
    test("stopFetching", () => {
        expect(uiActions.stopFetching()).toMatchSnapshot();
    });
    test("emitError with meta", () => {
        expect(uiActions.emitError(__.error, __.meta)).toMatchSnapshot();
    });
    test("emitError without meta", () => {
        expect(uiActions.emitError(__.error)).toMatchSnapshot();
    });
    test("setOnlineState", () => {
        expect(uiActions.setOnlineState()).toMatchSnapshot();
    });
    test("setOfflineState", () => {
        expect(uiActions.setOfflineState()).toMatchSnapshot();
    });
});
