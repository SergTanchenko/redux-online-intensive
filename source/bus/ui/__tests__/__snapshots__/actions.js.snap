// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions:  emitError with meta 1`] = `
Object {
  "error": true,
  "meta": Object {
    "prop1": "value1",
  },
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`ui actions:  emitError without meta 1`] = `
Object {
  "error": true,
  "meta": null,
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`ui actions:  setOfflineState 1`] = `
Object {
  "type": "SET_OFFLINE_STATE",
}
`;

exports[`ui actions:  setOnlineState 1`] = `
Object {
  "type": "SET_ONLINE_STATE",
}
`;

exports[`ui actions:  startFetching 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`ui actions:  stopFetching 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
